server/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -68,27 +67,39 @@ public Map<String, Set<ClusterBlock>> indices(ClusterBlockLevel level) {
     }
 
     private Set<ClusterBlock> blocksForIndex(ClusterBlockLevel level, String index) {
-        return indices(level).getOrDefault(index, emptySet());
+        return indices(level).getOrDefault(index, Set.of());
     }
 
     private static EnumMap<ClusterBlockLevel, ImmutableLevelHolder> generateLevelHolders(
         Set<ClusterBlock> global,
         Map<String, Set<ClusterBlock>> indicesBlocks
     ) {
         EnumMap<ClusterBlockLevel, ImmutableLevelHolder> levelHolders = new EnumMap<>(ClusterBlockLevel.class);
+        // reusable scratch list to collect matching blocks into in #addBlocksAtLevel temporarily, so we don't have to allocate it in the
+        // loop
+        List<ClusterBlock> scratch = new ArrayList<>();
```

Comment:
NIT: Should it have some initial capacity?

Reply:
Doesn't really matter I think, we only instantiate this once and while I suppose we could be tricky and make it `4` here or so, it probably isn't worth the extra complexity since it won't be a measurable change.